lib/js/jslib.js

```diff
@@ -1,187 +1,255 @@
 'use strict';
 
 /* LIST MANIPULATING FUNCTIONS */
+const LINKEDLIST =  new function(){
+  
+  return Object.freeze({
 
-var lsNil = null;
+    //Using [] here for easier parsing (if we used e.g. a string, JSON parsing would become ambiguous)
+    Nil: null,
```

Comment:
**Nitpick** why is `Nil` capitalised?

Reply:
It was like that before, I guess the original author wanted to mimick the "capitalize data constructors" rule found in languages like Haskell?

Reply:
I'm not sure if this is related, but some functions were capitalised at some point in order to ensure that there could never be a clash with user defined functions (which must always begin with lower case letters). There is a long outstanding bug with pattern matching compilation whereby one can witness strange behaviour by redefining hd and tl (collections.tests: Case patterns (with redefined hd)). It would make sense to address these name hygiene problems more systematically.